### PR TITLE
Add functionality for infinity norm of matrices

### LINKED ISSUE #13985: Implementation of infinity norm for matrices
The current _**norm()**_ method of the MatrixBase class doesn't have the implementation of infinity norm.

### PR DESCRIPTION
**Pull request by RavicharanN**
#### References to other Issues or PRs

Fixes #13985.

#### Brief description of what is fixed or changed

Functionality for the infinity norm (maximum of absolute row sums) has been added.
DocString has been modified and also the corresponding test has been added.

### CODE REVIEW

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -3452,6 +3452,14 @@ def norm(self, ord=None):
                 # Minimum singular value
                 return Min(*self.singular_values())
 
+            elif ord == S.Infinity:   # Infinity Norm - Maximum row sum
+                m = self.applyfunc(abs)
+                return Max(*[sum(m.row(i)) for i in range(m.rows)])
+
+            elif ord == S.NegativeInfinity: # Negative Infinity Norm - Maximum column sum
```
Comment (normalhuman) [inline sympy/matrices/matrices.py]: The "negative infinity norm" or matrices would not be maximum column sum. The maximum column sum is 1-norm, already implemented above.

If we want to introduce -oo norm for matrices, that merits a separate discussion. For example, [Matlab docs](https://www.mathworks.com/help/symbolic/norm.html?s_tid=gn_loc_drop) do not define it. 

Generally, one may think  that the order -p norm for matrices should be min||Ax||_p over ||x||_p = 1 (this is how it works for p=2 currently) but this is just one interpretation.

**Review by normalhuman — CHANGES_REQUESTED** [review]
The docstring of norm method should be edited to reflect the existence of inf-norm.

A test needs to be added too. 

The proposed addition of "-inf norm" needs some justification: what other CAS implement this norm, and how? Unlike positive-order norms, the negative orders lack universally accepted mathematical interpretation.

Also, please edit the PR text according to Wiki page [Writing pull request title and description](https://github.com/sympy/sympy/wiki/Development-workflow#writing-pull-request-title-and-description)

**Comment by RavicharanN** [comment]
Sure, I'll make the changes

**Comment by RavicharanN** [comment]
@normalhuman Can you please review it now?

**Comment by normalhuman** [comment]
The table of norms in the docstring of the method should have an entry for oo norm: see the line 3381.

**Comment by RavicharanN** [comment]
PR updated

**Review by normalhuman — APPROVED** [review]
Looks good. 

In the future, please follow the [Developer Workflow](https://github.com/sympy/sympy/wiki/Development-workflow), in particular: (1) create a branch for each feature (instead of committing to master), (2) follow GitHub rules for automatic closing of issues by writing "Fixes #NNNNN" or "Closed #NNNNN" in the text of a pull request.

**Pull request merged**